### DISCUSSION TITLE
run nerdctl on WSL2

### DISCUSSION THREAD

**Discussion by fzhan** [containerd/nerdctl#765] [Q&A] [answered]
Hi,

I've been looking around to instruction on how to run nerdctl on WSL2 (Ubuntu 20.04), I've installed genie (systemd) and restarted the environment. But running 'containerd-rootless-setuptool.sh install' gives me:
```
[ERROR] Needs systemd (systemctl --user)
```
and as root, 
```
systemctl enable --now containerd
Failed to enable unit: Connection reset by peer
```

I must be missing something, what did I miss?

**Comment by AkihiroSuda** [comment]
I don't know about WSL2, but probably you can just directly run `containerd-rootless.sh` without systemd

**Comment by fzhan** [comment]
Thank you @AkihiroSuda, I ran containerd-rootless.sh in wsl, here's what it says:
"
WARN[0000] The host root filesystem is mounted as "". Setting child propagation to "rslave" is not supported.
[rootlesskit:parent] error: failed to setup UID/GID map: newuidmap 309 [0 1000 1 1 100000 65536] failed: : exec: "newuidmap": executable file not found in $PATH
"

I'm just wondering because from the readme file, it says this is "known to work with wsl". Any indication on this would be really appreciated.

Best,

**Comment by AkihiroSuda** [comment]
> exec: "newuidmap": executable file not found in $PATH

https://command-not-found.com/newuidmap

**Comment by ramanNarasimhan77** [comment]
@fzhan  I was able to setup nerdctl on WSL2 by following steps in this post - https://medium.com/geekculture/move-away-from-docker-now-and-utilize-wsl2-systemd-genie-containerd-with-nerdctl-d5f729dcf227

**Comment by fzhan** [comment]
@AkihiroSuda thank you, I will go through the post @ramanNarasimhan77 and give it another go.

**Comment by fzhan** [comment]
@ramanNarasimhan77 worked like a charm!

However, now nerdctl image pull xxx gives:

FATA[0079] failed to extract layer sha256:bc73a95c0b43a6de24cacd4317f889ecebd2ee9afafa422204bc1ecfb65409e9: failed to mount /var/lib/containerd/tmpmounts/containerd-mount777138962: operation not permitted: unknown

**Comment by fzhan** [comment]
More question: how to access the servers in WSL2?

**Comment by fzhan** [comment]
> More question: how to access the servers in WSL2?

Found this: https://stackoverflow.com/questions/61002681/connecting-to-wsl2-server-via-local-network

So the port exposed needs to be configured one by one.

**Comment by AkihiroSuda** [comment, marked as answer]
Rancher Desktop now supports running nerdctl inside WSL2 easily: https://rancherdesktop.io/

**Comment by Anutrix** [comment]
Rancher Desktop now supports running nerdctl inside WSL2......
But containers started by nerdctl are not accessible on Windows side. Not sure if this a recent issue.

If the engine is **dockerd(moby)**, A webserver started on Windows Rancher Desktop can be accessed on **WSL2 browsers and Windows browsers**.
But if the engine is **nerdctl**, a webserver started on Windows Rancher Desktop can be accessed on **WSL2 browsers only and NOT Windows browsers**.

Is there a bug ticket for this?

**Reply by AkihiroSuda** [reply]
Please report to https://github.com/rancher-sandbox/rancher-desktop/issues

**Reply by Anutrix** [reply]
It happens only with nerdctl in Rancher Desktop not dockerd in Rancher Desktop so wasn't sure where to report.
Will report it there now.
